lib/chef/knife/cookbook_site_install.rb

```diff
@@ -90,7 +102,7 @@ def run
         upstream_file = File.join(@install_path, "#{@cookbook_name}.tar.gz")
 
         @repo.sanity_check
```

Comment:
should this also be protected by `unless config[:no_vendor]`?  and possibly `@repo` shouldn't be built unless we're vendoring (although scope...)